state-processor/build.gradle

```
@@ -0,0 +1,23 @@
+plugins {
+    id 'java-library'
+}
+
+group = 'com.ververica'
+mainClassName = 'com.ververica.flink.training.exercises.ReadRidesAndFaresSnapshot'
+
+// common set of dependencies
+dependencies {
+    // --------------------------------------------------------------
+    // Compile-time dependencies that should NOT be part of the
+    // shadow jar and are provided in the lib folder of Flink
+    // --------------------------------------------------------------
+    api "org.apache.flink:flink-java:${flinkVersion}"
+    api "org.apache.flink:flink-runtime-web_${scalaBinaryVersion}:${flinkVersion}"
+    api "org.apache.flink:flink-runtime_${scalaBinaryVersion}:${flinkVersion}"
+    api "org.apache.flink:flink-streaming-java_${scalaBinaryVersion}:${flinkVersion}"
+    api "org.apache.flink:flink-streaming-scala_${scalaBinaryVersion}:${flinkVersion}"
+    api "org.apache.flink:flink-state-processor-api_${scalaBinaryVersion}:${flinkVersion}"
```

```suggestion
    implementation "org.apache.flink:flink-state-processor-api_${scalaBinaryVersion}:${flinkVersion}"
```

The rest is imported from the `common` project already.
As for `implementation` vs. `api`: the latter is only available to java-library which is not necessary here (we already apply `java` and `application` globally). Technically, `implementation` is not correct, but for our setup with `flinkShadowJar` it is the right thing to do.

---

Ok.